### PR TITLE
Optimize Tiled example asset dependencies

### PR DESCRIPTION
**Pull request by rparrett**
I have a Tiled map with a single 16x16 tileset.

Prior to this PR, the Tiled example would create an identical `AssetPath` and add it as an asset dependency 256 times. I haven't looked into the details, but bevy was not handling this well as one would hope, and there seemed to be a noticeable slowdown.

This seems to resolve the issue.

### CODE REVIEW

**Review by StarArawn — APPROVED** [review]
Looks good! I'm holding off on merging this until the new renderer is merged. I'm thinking that I'll merge that shortly..

**Comment by StarArawn** [comment]
Can you rebase/merge this so I can get it merged in?

**Comment by rparrett** [comment]
Thanks for the poke. Rebased.

**Pull request merged**
